synapse/rest/media/v1/storage_provider.py

```diff
@@ -93,6 +93,11 @@ async def store_file(self, path: str, file_info: FileInfo) -> None:
         if file_info.server_name and not self.store_remote:
             return None
 
+        if file_info.url_cache:
```

Comment:
Curious why you chose to do this here instead of where we make the URL previews? My guess is to ensure we don't break this if we add another way of having URL previews?

Reply:
A little bit, but mostly for existing code. The call hierarchy gets pretty messy above `store_into_file` and I found it difficult to figure out which paths might touch the url cache. And I didn't want to thread an extra boolean through `_generate_thumbnails`.

The choice between `store_into_file` and here was mostly arbitrary, though it seems like this is where we do the existing filtering based on `FileInfo`.